### PR TITLE
Jenkins Changes For Additional Flows With Env File Addition And Other Fixes

### PR DESCRIPTION
**Pull request by mohanajuhi166**
**Changes Made For Jenkins Pipelines** : Adjustment for **PR** : https://github.com/microsoft/llmops-promptflow-template/pull/163

Changes include:

- Adding **"ENV_VARS"** loading and creation of .**env file**, loading it via python
- Small cosmetic changes for **yq tool** to use **".name"** to filter to flow's folder.
- Addition of CI & PR flows for chat_with_pdf and class_flows
- Changes in **platform_ci, platform_pr and platform_cd jenkins pipelines**

Tested webapp deployment with new changes 

Screenshot 2024-06-16 at 20 06 25

### CODE REVIEW

File: .jenkins/jobs/webapp_deployment [hunk 1]
```diff
@@ -4,46 +4,106 @@ pipeline {
...
+                        // Check if the .env file was created
+                        sh 'ls -l .env''
+
+                        // Optionally, load the .env file using Python
+                        def loadEnvScript = """
+import os
+from dotenv import load_dotenv
+
+# Load the .env file
+load_dotenv('.env')
```
Comment (machteldbogels) [inline .jenkins/jobs/webapp_deployment:64]: Nit: could this code block be at the same level as the rest of the script?

Comment (mohanajuhi166) [reply .jenkins/jobs/webapp_deployment:64]: @machteldbogels  the issue is the python code block , doesn't work if indented further. It throws an error due to indention issue. I checked with Jenkins pipeline linter and it's expected to be this way.
But, I will reverify this again and share a screenshot

**Review by machteldbogels — APPROVED** [review]

File: .jenkins/jobs/webapp_deployment (hunk 1 above)
Comment (mohanajuhi166) [reply .jenkins/jobs/webapp_deployment:64]: Screenshot 2024-06-17 at 18 21 05
Screenshot attached shows jenkins pipeline validation with the indentation in the PR

Comment (mohanajuhi166) [reply .jenkins/jobs/webapp_deployment:64]: If I change the indent to below , see error
Screenshot 2024-06-17 at 19 16 06
Screenshot 2024-06-17 at 19 15 56

File: .jenkins/pipelines/chat_with_pdf_ci_dev [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    // Trigger Platform CI Dev Pipeline
+    stages {
+        stage('Trigger Platform CI Dev Pipeline') {
+            steps {
+                script {
+                    build job: 'platform_ci_dev', parameters: [
+                        string(name: 'env_name', value: "${params.env_name}"),
+                        string(name: 'use_case_base_path', value: "${params.use_case_base_path}"),
+                        string(name: 'deployment_type', value: "${params.deployment_type}"),
+                        string(name: 'rg_name', value: "${RESOURCE_GROUP_NAME_dev}"),
```
[the remark this answers is not in the document]
Comment (mohanajuhi166) [reply .jenkins/pipelines/chat_with_pdf_ci_dev]: fixed

**Review by ritesh-modi — APPROVED** [review]
LGTM. Just check for conflicts

**Pull request merged**